File: Java/프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.java
class Solution {
    public long solution(int price, int money, int count) {
        long answer = money-f(price,count);
        
        return answer>0?0:-answer;
    }
    long f(long price,long count){
        return count==1?price:count*price+f(price,count-1);
        
    }
}

File: Kotlin/프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.kt
class Solution {
    fun solution(price: Int, money: Int, count: Int): Long {
        var answer: Long = -1
        var totalPrice = count.toLong()*(count+1)/2*price
        var needMoney = totalPrice-money
        return if(needMoney>0) needMoney else 0
    }
}